docs: Update changelog for changes in main. (#33577)
zulip/zulip: docs/overview/changelog.md

#### Highlights

- Redesigned how channel messages without a topic work to be much
  nicer, displayed as a special _general chat_ topic.
- Redesigned the left sidebar visually, adding convenient "New topic"
  and "New direct message" buttons, and allowing topics to line-wrap
  to 2 lines. The collapsed state of the direct messages section not
  persists across reloads.
- Redesigned the right sidebar buddy list visually, adding a new "THIS
  CONVERSATION" section showing just the users who've participated
  recently in the currently viewed conversation, and improving its
  filtering component.
- Replaced the compact mode setting with flexible options for font
  size and line spacing in the Zulip web app.
- User groups are now much more powerful, supporting subgroups and
  with new permissions settings for who can administer the group, join
  it, or edit its membership. Permission for creating new groups is
  now separate from permission to administer all groups.

> - Invitation links can now encode initial groups for a user.

Reviewer: ```suggestion
- Invitation links and emails can now encode initial groups for a user.
```

- Most permissions settings in Zulip have been reimplemented with a
  new flexible system based on groups, and can have a value of any
  combination of roles, groups, and individual users. Groups can be
  nested as subgroups of other groups. Groups are now deactivated,
  rather than being fully deleted, so Zulip's audit logs still
  maintain a complete history of permissions changes.

> - Group settings was redesigned, with many new administration

Reviewer: ```suggestion
- Group settings were redesigned, with many new administration
```

  permissions, and a new Permissions panel showing all the permissions
  that membership in the group grants.
- Channel administration permissions have been restructured to be
  based on groups, with new settings for administrering channels,
  adding subscribers, removing them, and being able to join channels.
- The access control model for message content in channels has been
  generalized to allow users to be able to join a private channel at
  their convenience, without needing to be subscribed. Users who have
  permission to subscribe themselves to a channel are able to preview
  and search channel content, just like how non-guest users can access
  public channels without first subscribing.
- Added a right sidebar buddy list style option showing user avatars,

>   designed for smaller organizations. You can now change the buddy

Reviewer: I love it on CZO, so :
```suggestion
You can now change the buddy
```
(And then , -> . on the line before.)

  list style directly from a menu in the sidebar.
- Changes the semantics for when messages are marked as read when
  scrolling the blue selection box past them and the bottom of the
  feed is not visible to avoid shifting one's place when briefly
  visiting a conversation.
- Reworked the "archive channel" functionality to not mutate their
  names or always make content inaccessible, with a view towards

>   supporting unarchiving channels.

Reviewer: - Reworked the "archive channel" functionality to preserve channel names and content access, with a view...

  will need to updated for most self-hosted installations; see the
  upgrade notes below.
- Added a new server-to-server Zoom integration, which is much more
  convenient to set up with current Zoom marketplace policies. The
  BigBlueButton call provider integration now supports voice-only
  calls.
- Added an automated flow for transferring the mobile push
  notifications service registration for a given domain to a new
  server, even if the secrets have been lost.
- Added new typing indicators for when someone is editing a message
  that you're viewing in the message feed.

#### Full feature changelog

- Added installer options for automatically registering for the mobile
  push notifications service.
- Added support for syncing user roles with the SCIM integration.
- Added support for configuring certain custom profile fields to not
  be editable by the user, for fields that are maintained by
  administrators or synced from a third-party service.
- Added support for personal Microsoft accounts in the EntraID
  (formerly AzureAD) authentication backend. Updated documentation to
  consistently refer to this authentication method as Microsoft Entra

>   ID.

Reviewer: So "Entra ID" or "EntraID"? Pick one. ;)

> - Added a new "Forward message" message in the message actions menu,

Reviewer: ```suggestion
- Added a new "Forward message" option in the message actions menu,
```

  similar to "Quote message", but with an intent to send the message
  elsewhere.
- Added support for copy/pasting and quoting selections containing
  LaTeX.
- Added `*` keyboard shortcut for navigating to starred messages view.
- Added new personal and organization-level settings controlling
  access to and display of any future AI features.
- Added UI support for data exports with member consent, including a
  privacy setting for users to specify whether their private data can
  be included in data exports.
- Added new prompt when editing a message to remove the last reference
  to a previously uploaded file.
- Added new prompt when your timezone doesn't match your profile
  timezone. The prompt can be disabled via a setting.
- Added a new configurable warning when composing a direct message
  including a guest as a recipient.
- Added the "Enter sends" setting to the Preferences settings
  panel. Previously, it was only accessible in the compose area.

> - Added a wizard for creating collaborative TODO widgets.

Reviewer: ```suggestion
- Added a wizard for creating collaborative to-do widgets.
```

- Fixed issues with outgoing webhook bots using the API to interact
  with messages in private channels.
- Reimplemented the setting to demote inactive channels in the left
  sidebar, fixing an issue that could cause the sidebar to constantly
  rerender while message history was loaded.
- Fixed the left sidebar sometimes not scrolling properly when
  navigating to a topic.
- Fixed several subtle race bugs involving local echo of sent messages.
- Fixed several subtle bugs involving the compose preview area's height.
- Fixed languages with custom playgrounds not being offered in the
  default code block languages UI.
- Fixed an issue that could cause browser/desktop app windows to
  reload before a previously suspended laptop had gained reliable
  access to the network.
- Fixed buggy URL-escaping of filenames in tooltips.
- Fixed bugs preventing reliably rolling restart of server processes.
- Fixed several issues with the "load more" button in the recent view.
- Fixed handling of several rare race conditions.
- Fixed animated images incorrectly autoplaying on Firefox, even when
  the user has configured them to only play on hover.

> - Fixed a bug that allowed the creation of invalid linkifirs.

Reviewer: ```suggestion
- Fixed a bug that allowed the creation of invalid linkifiers.
```
